extension/src/Telemetry.ts

```diff
@@ -0,0 +1,43 @@
+import * as SettingsLoader from "./Settings/SettingsLoader";
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+import appInsights = require("applicationinsights");
+const enableTelemetry = SettingsLoader.getSettings().enableTelemetry;
```

Comment:
Doesn't this require a reload for a change in the setting to take effect? That's probably a good thing performance wise.  We might want to specify that in the setting description and README.

Reply:
That is **if** that is the behavior.

Reply:
That's by design, since we want the telemetry to have as small performance impact as possible. I'll document that in the setting.

Reply:
done